go/consensus/tendermint/apps/roothash/slashing.go

```diff
@@ -43,11 +44,155 @@ func onEvidenceRuntimeEquivocation(
 		return fmt.Errorf("tendermint/roothash: nothing to slash from account %s", entityAddr)
 	}
 
-	// Move slashed amount to the runtime account.
-	// TODO: part of slashed amount (configurable) should be transferred to the transaction submitter.
+	// If the caller is a node, distribute slashed funds to the controlling entity instead of the
+	// caller directly.
+	rewardAddr := ctx.CallerAddress()
```

Comment:
I wonder if people frontrun transactions like these, to take the reward for themselves

Reply:
Yeah that's a valid concern. Incentivizing evidence submission seems like it introduces more problems than it solves :-)